### ISSUE TITLE
Blog widgets (e.g. archive widget) are not specific to the blog they're created on

### ISSUE THREAD

**Issue by robbieaverill** [silverstripe/silverstripe-blog#581]
**Blog 3.3.x-dev**

Example:

* Create Blog A
* Add Blog Post A to it
* Add a content and archive widget to Blog A (uncheck "inherit from parent")
* Publish both
* Create Blog B
* Add Blog Post B to it
* Add a content and archive widget to Blog A (uncheck "inherit from parent")
* Publish both
* Archive Blog A
* Go to Blog B on the frontend
* Click on the current month and day (in the blog archive widget)

It shows no results, but it should show the Blog Post B entry. This happens because the archive returned is the one originally created in Blog Post A, but it's being rendered in Blog Post B.

If you delete the widgets now from Blog B and add them again, you then see the correct behaviour on the frontend.

**Comment by robbieaverill** [comment]
Reduced impact - the blog for the widget is actually configurable in the CMS (which I hadn't noticed). It should default to the blog you add it to though, rather than one of the other ones which is what happened in my case.

**Comment by BlinkTwice** [comment]
I'm experiencing this too but the 'reduced impact' fix hasn't resolved it for me. I tested the following code in the [BlogArchiveWidget](https://github.com/silverstripe/silverstripe-blog/blob/a8b42b151a43ea26ae132b6f0545f2c25c094f72/src/Widgets/BlogArchiveWidget.php#L128) which did work for me:
```
if ((int)$this->BlogID) {
    $query->addInnerJoin('SiteTree' . $suffix,
        '"SiteTree' . $suffix . '"."ID" = "BlogPost' . $suffix . '"."ID"')
        ->addSelect('"SiteTree' . $suffix . '"."ParentID" AS "ParentID"')
        ->addWhere(['"ParentID" = ?' => (int)$this->BlogID]);
}
```
Happy to PR if suitable solution.

**Comment by robbieaverill** [comment]
@BlinkTwice thanks, a pull request would be welcomed. Can you make it against the 3.2 or 3.3 branch?

**Comment by BlinkTwice** [comment]
This is actually fixed in 3.3 with 01c1ead069825c383fd31e72b35d835da65be558

**Comment by robbieaverill** [comment]
Cool, thank you for investigating and reporting the issue
